askama_parser/src/lib.rs

```diff
@@ -350,13 +350,130 @@ fn str_lit(i: &str) -> ParseResult<'_> {
     Ok((i, s.unwrap_or_default()))
 }
 
+/// Represents the different kinds of char declarations:
+enum Char<'a> {
+    /// Any character that is not escaped.
+    Literal,
+    /// An escaped character (like `\n`) which doesn't require any extra check.
+    Escaped,
+    /// Ascii escape (like `\x12`).
+    AsciiEscape(&'a str),
+    /// Unicode escape (like `\u{12}`).
+    UnicodeEscape(&'a str),
+}
+
+impl<'a> Char<'a> {
+    fn parse(i: &'a str) -> ParseResult<'a, Self> {
```

Comment:
Make this a `FromStr` impl?

Reply:
It requires to implement:

```rust
fn from_str(s: &str) -> Result<Self, Self::Err> {
```

So it's gonna be more trouble than it's worth imo.